apps/src/templates/teacherNavigation/TeacherNavigationBar.tsx

```diff
@@ -70,11 +70,34 @@ const TeacherNavigationBar: React.FC<{
     );
   };
 
+  const navigate = useNavigate();
+  const location = useLocation();
+  const urlSectionId = useParams().sectionId;
+
+  const [currentPathName, currentPathObject] = React.useMemo(() => {
+    return (
+      _.find(
+        Object.entries(LABELED_TEACHER_NAVIGATION_PATHS),
+        path => !!matchPath(path[1].absoluteUrl, location.pathname)
+      ) || [null, null]
+    );
+  }, [location]);
+
+  React.useEffect(() => {
+    if (urlSectionId && parseInt(urlSectionId) !== selectedSection?.id) {
+      asyncLoadSelectedSection(urlSectionId);
+    }
+  }, [urlSectionId, selectedSection?.id]);
+
   const coursecontentSectionTitle = getSectionHeader(i18n.courseContent());
 
   let courseContentKeys: (keyof typeof LABELED_TEACHER_NAVIGATION_PATHS)[];
   if (selectedSection?.unitName) {
-    courseContentKeys = ['unitOverview', 'lessonMaterials', 'calendar'];
+    if (currentPathName === TEACHER_NAVIGATION_PATH_NAMES.nestedUnitOverview) {
```

Comment:
What's the plan to get rid of the non-nested unitOverview?

Reply:
I'm thinking that once the project is live and stable, we will start remove support for the deprecated paths.